webapp/components/MasonryGrid/MasonryGridItem.vue

```diff
@@ -5,15 +5,33 @@
 </template>
 
 <script>
+const landscapeRatio = 1.3
+const squareRatio = 1
+const portraitRatio = 0.7
+
+const getRowSpan = aspectRatio => {
+  if (aspectRatio >= landscapeRatio) return 13
+  else if (aspectRatio >= squareRatio) return 15
+  else if (aspectRatio >= portraitRatio) return 18
+  else return 25
+}
```

Comment:
This whole thing is a hack but it works ok for setting up the initial layout. If there's a math whizz around who has an idea for an actual `formula` to map these values I'd appreciate it. :wink:

Reply:
I don't know myself, but how about https://github.com/desandro/masonry? It's JavaScript not CSS but it looks quite small: https://github.com/desandro/masonry/blob/master/masonry.js

Reply:
Does that help maybe: https://w3bits.com/css-grid-masonry/ ?